nf_core/refgenie.py

```diff
@@ -0,0 +1,146 @@
+        for asset in asset_list:
+            try:
+                pth = rgc.seek(genome, asset)
+            # Catch general exception instead of refgencof exception --> no refgenconf import needed
+            except Exception as e:
+                log.warn(f"{genome}/{asset} is incomplete, ignoring...")
+            else:
+                genomes_str += '      {} = "{}"\n'.format(asset.ljust(20, " "), pth)
+
+    return NF_CFG_TEMPLATE.format(content=genomes_str)
+
+
+def _update_nextflow_home_config(refgenie_genomes_config_file, nxf_home):
+    """
+    Update the $NXF_HOME/config file by adding a includeConfig statement to it
+    for the 'refgenie_genomes_config_file' if not already defined
+    """
+    # Check if NXF_HOME/config exists and has a
+    include_config_string = f"includeConfig '{os.path.abspath(refgenie_genomes_config_file)}'\n"
+    nxf_home_config = os.path.join(nxf_home, "config")
+    if os.path.exists(nxf_home_config):
+        # look for include statement in config
+        has_include_statement = False
+        with open(nxf_home_config, "r") as fh:
+            lines = fh.readlines()
+            for line in lines:
+                if re.match(f"\s*includeConfig\s*'{os.path.abspath(refgenie_genomes_config_file)}'", line):
+                    has_include_statement = True
+                    break
+
+        # if include statement is missing, add it to the last line
+        if not has_include_statement:
+            with open(nxf_home_config, "a") as fh:
+                fh.write(include_config_string)
+
+            log.info(f"Included refgenie_genomes.config to {nxf_home_config}")
+
+    else:
+        # create new config and add include statement
+        with open(nxf_home_config, "w") as fh:
+            fh.write(include_config_string)
+            log.info(f"Created new nextflow config file: {nxf_home_config}")
+
+
+def update_config(rgc):
+    """
+    Update the genomes.config file after a local refgenie database has been updated
+
+    This function is executed after running 'refgenie pull <genome>/<asset>'
+    The refgenie config file is transformed into a nextflow.config file, which is used to
+    overwrited the 'refgenie_genomes.config' file.
+    The path to the target  config file is inferred from the following options, in order:
+
+    - the 'nextflow_config' attribute in the refgenie config file
+    - the NXF_REFGENIE_PATH environment variable
+    - otherwise defaults to: $NXF_HOME/nf-core/refgenie_genomes.config
+
+    Additionaly, a 'includeConfig' statement is added to the file $NXF_HOME/config
+    """
+
+    # Compile nextflow refgenie_genomes.config from refgenie config
+    refgenie_genomes = _print_nf_config(rgc)
+
+    # Get the path to NXF_HOME
+    # If NXF_HOME is not set, create it at $HOME/.nextflow
+    # If $HOME is not set, set nxf_home to false
+    nxf_home = os.environ.get("NXF_HOME")
+    if not nxf_home and "HOME" in os.environ:
```

Comment:
Might be nice to refactor this code a bit more now that we're using `Pathlib` more. Can probably make it a lot shorter / more readable by using `Path.home()`: https://docs.python.org/3/library/pathlib.html#pathlib.Path.home